### PR TITLE
docker-compose version 2

### PR DESCRIPTION
**Pull request by thiagotalma**
@dsander v2 all files done!

### CODE REVIEW

File: docker/single-process/docker-compose.yml [hunk 1]
```diff
@@ -1,35 +1,39 @@
...
-    - mysql
+  mysql:
+    image: mysql:5.7
+    volumes_from:
+      - mysqldata
+    environment:
+      MYSQL_ROOT_PASSWORD: myrootpassword
+      MYSQL_DATABASE: huginn
+      MYSQL_USER: huginn
+      MYSQL_PASSWORD: myhuginnpassword
```
Comment (dsander) [inline docker/single-process/docker-compose.yml]: compose does not set the link environment variables automatically anymore, we need to add them for all huginn processes:
```
      MYSQL_PORT_3306_TCP_ADDR: mysql
      MYSQL_PORT_3306_TCP_PORT: 3306
```

File: docker/single-process/postgresql.yml [hunk 2]
```diff
@@ -1,72 +1,76 @@
...
-    - postgres
-  command: /scripts/init bin/threaded.rb
+  huginn_web:
+    image: cantino/huginn-single-process
+    restart: always
+    extends:
+      file: environment.yml
+      service: huginn_base
+    environment:
+      DATABASE_ADAPTER: postgresql
```
Comment (dsander) [inline docker/single-process/postgresql.yml]: same here (and for all container in this file)
```
      POSTGRES_PORT_5432_TCP_ADDR: postgres
      POSTGRES_PORT_5432_TCP_PORT: 5432
```

File: docker/single-process/docker-compose.yml [hunk 3]
```diff
@@ -1,35 +1,39 @@
...
-  links:
-    - mysql
-  command: /scripts/init bin/threaded.rb
+  huginn_web:
+    image: cantino/huginn-single-process
+    restart: always
+    extends:
+      file: environment.yml
+      service: huginn_base
+    ports:
```
Comment (dsander) [inline docker/single-process/docker-compose.yml]: compose does not set the link environment variables automatically anymore, we need to add them for all huginn processes:
```
    environment:
      MYSQL_PORT_3306_TCP_ADDR: mysql
      MYSQL_PORT_3306_TCP_PORT: 3306
```

**Comment by dsander** [comment]
@thiagotalma Thanks!

File: docker/single-process/docker-compose.yml (hunk 3 above)
Comment (thiagotalma) [inline docker/single-process/docker-compose.yml]: @dsander done!

File: docker/single-process/environment.yml [hunk 4]
```diff
@@ -1,7 +1,13 @@
...
+
+services:
+  huginn_base:
+    environment:
+      DATABASE_ADAPTER: mysql2
+      DATABASE_NAME: huginn
+      DATABASE_USERNAME: huginn
+      DATABASE_PASSWORD: myhuginnpassword
+      APP_SECRET_TOKEN: 3bd139f9186b31a85336bb89cd1a1337078921134b2f48e022fd09c234d764d3e19b018b2ab789c6e0e04a1ac9e3365116368049660234c2038dc9990513d49c
+      MYSQL_PORT_3306_TCP_ADDR: mysql
```
Comment (dsander) [inline docker/single-process/environment.yml]: The mysql specific environment variables can not be in here, the break the container setup of `postgresql.yml`.

**Comment by dsander** [comment]
Works great, thanks @thiagotalma!

**Pull request merged**